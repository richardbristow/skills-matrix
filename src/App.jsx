import React, { useState, useEffect } from 'react';
import styled from 'styled-components/macro';
import { Auth } from 'aws-amplify';

import Routes from './Routes';
import SideBar from './components/SideBar';
import AuthenticatedUserContext from './AuthenticatedUserContext';

const StyledApp = styled.div`
  display: grid;
  grid-template-areas: 'sidebar main';
  grid-template-columns: 250px 1fr;
  height: 100vh;
`;

const App = () => {
  const [authenticated, setAuthenticated] = useState(false);
  const [authenticating, setAuthenticating] = useState(true);
  const [authenticatedUser, setAuthenticatedUser] = useState({});

  const handleLogout = async ({ history }) => {
    await Auth.signOut();
    setAuthenticated(false);
    setAuthenticatedUser({});
    history.push('/login');
  };

  useEffect(() => {
    const getUserSession = async () => {
      try {
        const {
          accessToken: {
            payload: { 'cognito:groups': group },
          },
          idToken: {
            payload: { name },
          },
          idToken: {
            payload: { email },
          },
          accessToken: {
            payload: { username },
          },
        } = await Auth.currentSession();
        setAuthenticatedUser({ name, email, username, group: group[0] });
        setAuthenticated(true);
      } catch (error) {
        if (error !== 'No current user') {
          // eslint-disable-next-line no-console
          console.error(error);
        }
      }
      setAuthenticating(false);
    };
    getUserSession();
  }, [authenticating]);

  return (
    !authenticating && (
      <AuthenticatedUserContext.Provider value={authenticatedUser}>
        <StyledApp>
          <SideBar authenticated={authenticated} handleLogout={handleLogout} />
          <main>
            <Routes
              authenticated={authenticated}
              setAuthenticating={setAuthenticating}
              setAuthenticated={setAuthenticated}
            />
          </main>
        </StyledApp>
      </AuthenticatedUserContext.Provider>
    )
  );
};

export default App;
