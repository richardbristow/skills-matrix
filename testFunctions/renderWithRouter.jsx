import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';

const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history,
  };
};

export default renderWithRouter;
