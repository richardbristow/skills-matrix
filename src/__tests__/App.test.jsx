import React from 'react';
import { waitForDomChange, getByText } from '@testing-library/react';
import App from '../App';
import renderWithRouter from '../../testFunctions/renderWithRouter';

describe('App', () => {
  it('renders without throwing errors', async () => {
    const { container } = renderWithRouter(<App />);
    await waitForDomChange();
    expect(
      getByText(container, 'Login', { selector: 'h3' }),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
