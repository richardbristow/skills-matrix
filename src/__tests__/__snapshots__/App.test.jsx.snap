// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without throwing errors 1`] = `
<div>
  <div
    class="App__StyledApp-n5nkjo-0 fLlJrz"
  >
    <aside
      class="SideBar__StyledSideBar-sc-1l67eb2-0 lldNPQ"
    >
      <ul
        class="SideBar__StyledSideBarMenu-sc-1l67eb2-1 lbJvnL"
      >
        <li
          class="SideBar__StyledSidebarMenuLink-sc-1l67eb2-3 bEKnlD"
        >
          <a
            _css2="
        cursor: ,default,;
      "
            class="SideBar__StyledSideBarLink-sc-1l67eb2-2 sc-fjdhpX pdQzA"
            href="/"
          >
            <h3>
              Skills Matrix
            </h3>
          </a>
        </li>
        <li
          class="SideBar__StyledSidebarMenuLink-sc-1l67eb2-3 ysUGI"
        >
          <a
            _css2="
        cursor: ,pointer,;
      "
            aria-current="page"
            class="SideBar__StyledSideBarLink-sc-1l67eb2-2 sc-fjdhpX iiRcAn active"
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </aside>
    <main>
      <div
        class="StyledMain-sc-1o4u4y0-0 Login__StyledLogin-sc-1bf8mvn-0 kcTDFU"
      >
        <div
          class="card"
        >
          <h3
            class="card-header"
          >
            Login
          </h3>
          <div
            class="card-body"
          >
            <form
              class=""
              novalidate=""
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="email"
                >
                  Email address
                </label>
                <div
                  class="input-group"
                >
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      <svg
                        fill="none"
                        height="16"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          cx="12"
                          cy="12"
                          r="4"
                        />
                        <path
                          d="M16 8v5a3 3 0 0 0 6 0v-1a10 10 0 1 0-3.92 7.94"
                        />
                      </svg>
                    </span>
                  </div>
                  <input
                    class="form-control"
                    id="email"
                    name="email"
                    placeholder="Enter email..."
                    required=""
                    type="email"
                    value=""
                  />
                  <div
                    class="invalid-feedback"
                  >
                    Please provide a valid email address.
                  </div>
                </div>
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="password"
                >
                  Password
                </label>
                <div
                  class="input-group"
                >
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      <svg
                        fill="none"
                        height="16"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          height="11"
                          rx="2"
                          ry="2"
                          width="18"
                          x="3"
                          y="11"
                        />
                        <path
                          d="M7 11V7a5 5 0 0 1 10 0v4"
                        />
                      </svg>
                    </span>
                  </div>
                  <input
                    class="form-control"
                    id="password"
                    name="password"
                    placeholder="Password..."
                    required=""
                    type="password"
                    value=""
                  />
                  <div
                    class="invalid-feedback"
                  >
                    Please provide a password.
                  </div>
                </div>
              </div>
              <button
                class="btn btn-outline-primary btn-block"
                css="cursor: not-allowed;"
                disabled=""
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
